### DISCUSSION TITLE
serial_groups across pipelines

### DISCUSSION THREAD

**Discussion by hsiliev** [concourse/concourse#8863] [Help & Support]
I'm having a VM in GCP and trying to prevent jobs in different pipelines making changes at the same time.

I have a serial_groups in place:
```
serial_groups: [ gcp ]
```

Noticed however that `serial_groups` do not stop jobs in another pipeline. Is this the case? Nothing I can find in the documentation about the scope of the serial_groups "locks"

**Comment by marco-m-pix4d** [comment]
Hello,
the scope of a serial group are jobs in the same pipeline.

On the other hand, I think that you can achieve what you are looking for with something like https://github.com/concourse/pool-resource. It sort of works. If you go down this route, you might consider also https://github.com/Pix4D/concourse-pool-boy.

If you have at your disposal something like Consul, I would use it instead of the pool-resource, see (https://developer.hashicorp.com/consul/docs/dynamic-app-config/sessions).

**Comment by hsiliev** [comment]
Thanks @marco-m-pix4d! 

Indeed we're already using pool-resource, but noticed the same behaviour as you did with locks left claimed for several reasons. Your reference to pool-boy is much appreciated.

Serial groups could really replace both pool-resource and external systems if their scope could be larger than a pipeline. In the end their management is left to Concourse that can couple this with pipeline, job and task management as needed.
